src/components/NumberInput/index.js

```diff
@@ -0,0 +1,212 @@
+            boxShadow: 'inset 2px 4px 4px 1px rgb(85 108 165 / 56%)',
+          })
+        }
+        onMouseUp={() => setDownMouseClickStyle({})}
+        onMouseOut={() => setDownMouseClickStyle({})}
+        style={downMouseClickStyle}
+        disabled={props.selectedScreenSaver === '(None)'}
+      />
+      <input
+        type="number"
+        className={
+          props.selectedScreenSaver === '(None)'
+            ? 'numberInputDisabled'
+            : 'numberInput'
+        }
+        ref={inputRef}
+        value={inputValue}
+        onChange={e => checkAndSetInput(e.target.value)}
+        onKeyDown={handleKeyDown}
+        onKeyUp={handleKeyUp}
+        onFocus={e => e.target.select()}
+        disabled={props.selectedScreenSaver === '(None)'}
+      />
+    </StyledNumberInput>
+  );
+}
+
+const StyledNumberInput = styled.div`
+  .numberInput {
+    border-radius: 0;
+    border: 1px solid gray;
+    font-size: 13px;
+    width: 49px;
+    height: 21px;
+    padding-bottom: 2px;
+    padding-right: 2px;
+    text-align: right;
+    &:focus {
+      outline: transparent;
+    }
+    &::selection {
+      background: #2f71cd;
+      color: white;
+    }
+  }
+
+  .numberInputDisabled {
+    border-radius: 0;
+    font-size: 13px;
+    width: 49px;
+    height: 21px;
+    padding-bottom: 2px;
+    padding-right: 2px;
+    color: #adaa9c;
+    border: 1px solid #afafaf;
+    text-align: right;
+  }
+
+  .arrowUp {
+    color: inherit;
+    border: none;
+    padding: 0;
+    font: inherit;
+    cursor: pointer;
+    outline: inherit;
+    position: absolute;
+    top: 38px;
+    left: 69px;
+    width: 15px;
+    height: 9px;
+    transform: rotate(180deg);
+    background-image: url(${numberDown});
+    z-index: 1;
+    &:hover {
+      filter: brightness(1.07);
+    }
+  }
+
+  .arrowDown {
+    background: none;
+    color: inherit;
+    border: none;
+    padding: 0;
+    font: inherit;
+    cursor: pointer;
+    outline: inherit;
+    position: absolute;
+    position: absolute;
+    top: 46.5px;
+    left: 69px;
+    width: 15px;
+    height: 9px;
+    background-image: url(${numberDown});
+    z-index: 1;
+    &:hover {
+      filter: brightness(1.07);
+    }
+  }
+
+  .disabledArrowUp {
+    color: inherit;
+    border: none;
+    padding: 0;
+    font: inherit;
+    cursor: inherit;
+    outline: inherit;
+    position: absolute;
+    top: 38px;
+    left: 69px;
+    width: 15px;
+    height: 9px;
+    transform: rotate(180deg);
+    background-image: url(${numberDown});
+    filter: grayscale(100%);
+    opacity: 0.3;
+    z-index: 1;
+  }
+
+  .disabledArrowDown {
+    color: inherit;
+    border: none;
+    padding: 0;
+    font: inherit;
+    cursor: inherit;
+    outline: inherit;
+    position: absolute;
+    position: absolute;
+    top: 47px;
+    left: 69px;
+    width: 15px;
+    height: 9px;
+    background-image: url(${numberDown});
+    filter: grayscale(100%);
+    opacity: 0.3;
+    z-index: 1;
+  }
```

Comment:
Lots of duplication here

Reply:
Combined disabled & arrow classes to one styled component